src/main/java/com/health/controller/HomeController.java

```diff
@@ -1348,6 +1348,184 @@ private static String getLanguageCode(String languageName) {
+                .findByid(tutorial.getConAssignedTutorial().getTopicCatId().getCat().getCategoryId());
+        List<TopicCategoryMapping> topicCatMapping = topicCatService.findAllByCategory(category);
+        List<ContributorAssignedTutorial> contriAssignedTut = conRepo.findAllByTopicCat(topicCatMapping);
+        List<Tutorial> tutorials = tutService.findAllByContributorAssignedTutorialList1(contriAssignedTut);
+
+        for (Tutorial x : tutorials) {
+            if (x == tutorial) {
+                continue;
+            }
+
+            if (x.getConAssignedTutorial().getLan().getLangName()
+                    .equalsIgnoreCase(tutorial.getConAssignedTutorial().getLan().getLangName())) {
+                relatedTutorial.add(x);
+            }
+
+        }
+
+        Collections.sort(relatedTutorial, Tutorial.UserVisitComp);
+
+        model.addAttribute("tutorials", relatedTutorial);
+
+        Set<String> catTemp = new HashSet<String>();
+        Set<String> topicTemp = new HashSet<String>();
+        Set<String> lanTemp = new HashSet<String>();
+
+        List<Tutorial> tutorialse = tutService.findAllByStatus(true);
+        for (Tutorial temp : tutorialse) {
+            ContributorAssignedTutorial conAssignedTutorial = temp.getConAssignedTutorial();
+            catTemp.add(conAssignedTutorial.getTopicCatId().getCat().getCatName());
+            lanTemp.add(conAssignedTutorial.getLan().getLangName());
+            topicTemp.add(conAssignedTutorial.getTopicCatId().getTopic().getTopicName());
+        }
+
+        getModelData(model, catId, topicId, lanId, query);
+        List<Integer> scriptVerList = getApiVersion(scriptmanager_api, category.getCategoryId(),
+                tutorial.getTutorialId(), lanName.getLanId());
+        String sm_url = "";
+        String sm_url2 = "";
+        StringBuilder sb = new StringBuilder();
+
+        sb.append(scriptmanager_url);
+        sb.append(scriptmanager_path);
+        sb.append(String.valueOf(category.getCategoryId()));
+        sb.append("/");
+        sb.append(String.valueOf(tutorial.getTutorialId()));
+        sb.append("/");
+        sb.append(String.valueOf(lanName.getLanId()));
+        sb.append("/");
+        sb.append(String.valueOf(tutorial.getConAssignedTutorial().getTopicCatId().getTopic().getTopicName()));
+        sb.append("/");
+        if (scriptVerList != null && scriptVerList.size() > 0) {
+
+            StringBuilder sb2 = new StringBuilder(sb);
+            sb2.append(scriptVerList.get(0));
+            sm_url = sb2.toString();
+        }
+
+        if (principal != null) {
+
+            boolean flag = checkRole(usr, category, lanName);
+            if (flag) {
+                StringBuilder sb1 = new StringBuilder(sb);
+                sb1.append('2');
+                sm_url2 = sb1.toString();
+
+            }
+        }
+
+        model.addAttribute("sm_url", sm_url);
+        model.addAttribute("sm_url2", sm_url2);
+        return "oldTutorial";

```

Comment:
Replace this code with

retrurn viewTutorial(req, cat, topic, lan, 0, 0, 0, query, principal, model);